### DISCUSSION TITLE
Override the existing snip?

### DISCUSSION THREAD

**Discussion by Weissle** [L3MON4D3/LuaSnip#589] [Q&A] [answered]
I use [friendly-snippets](https://github.com/rafamadriz/friendly-snippets) and myself snippet.
How can I override some snippets in the `friendly-snippets`? For example, if I define my `for` snippet and I will see two `for` snippets after that.

Another question: How can I let LuaSnip search snippet in the runtime path and load the snippet from a specific path at the same time? For now, to load them both, I use the below code.
```lua
-- load friendly-snippets
require("luasnip.loaders.from_vscode").lazy_load({})
-- load my snippets
require("luasnip.loaders.from_vscode").lazy_load({
    paths = { "./snippets" },
})
```
Thanks for your reply in advance.

**Comment by L3MON4D3** [comment, marked as answer]
> I use friendly-snippets and myself snippet.
How can I override some snippets in the friendly-snippets? For example, if I define my for snippet and I will see two for snippets after that.

Some interesting discussion on that in #458, I'll have to get to implementing that eventually

> Another question: How can I let LuaSnip search snippet in the runtime path and load the snippet from a specific path at the same time? For now, to load them both, I use the below code.

Ouh, that's not possible :/
One solution would be allowing a special string passed to `paths`(`{paths = "RUNTIMEPATH"}` maybe), but I'm not sure if that's really the best way:

The loaders accept parameters like `exclude` (to exclude a certain filetype) and `include` (to exclude all except this filetype), and the `priority`, which is applied to all snippets loaded via that call.
These parameters, I think, only make sense per-snippet-collection (example: I'd like to load `lua`-snippets with prio 999 from friendly-snippets, and my entire personal repo with priority 1001).
So the problem is that we have, on the one hand, parameters you'd pass per-collection (at least applying them to multiple collections does not make much sense imo), but on the other hand we accept multiple collections (`paths` is a list). This is inconsistent, and I think we should make each call accept only one path.

So TLDR: reconsider loader-design. (although, the special string for runtimepath would be complementary to refactoring loaders... maybe it makes sense to add it)

**Reply by Weissle** [reply]
**Duplicated snippets**: We can filter the low-priority snippets out easily. For those snippets that have the same highest priority, we can accept all of them. By adjusting snippets' priority, users can decide which snippets to stay. For those snippets that come from LSP, users can disable them by changing `nvim-lspconfig`'s or their completion plugin's configuration, while `LuaSnip` can do nothing.

**Load from runtime path and specific path**: Maybe add an option for it? It should be easy. After all, the main function is finished. But  this feature is not necessary since users can call `load` or `lazy_load` twice or more.
